src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -47,24 +49,17 @@ public sealed class RestoreManagerPackage : AsyncPackage
             CancellationToken cancellationToken, 
             IProgress<ServiceProgressData> progress)
         {
-            var componentModel = await this.GetComponentModelAsync();
-            componentModel.DefaultCompositionService.SatisfyImportsOnce(this);
+            var componentModel = await GetServiceAsync(typeof(SComponentModel)) as IComponentModel;
+            componentModel.DefaultCompositionService.SatisfyImportsOnce(this);            
 
-            await SolutionManager.InitializeAsync(this);
-            await SolutionRestoreWorker.InitializeAsync(this);
-
-            await ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
-            {
-                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                var dte = await this.GetDTEAsync();
-                _buildEvents = dte.Events.BuildEvents;
-                _buildEvents.OnBuildBegin += BuildEvents_OnBuildBegin;
-
-                UserAgent.SetUserAgentString(
-                    new UserAgentStringBuilder().WithVisualStudioSKU(dte.GetFullVsVersionString()));
-            });
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Trying out multiple things, this is still not finalized code. Although this one left out by mistake, thanks for pointing!